### ISSUE TITLE
System.Diagnostics.DiagnosticSource 4.6.0-preview.x

### ISSUE THREAD

**Issue by mitchdenny** [Azure/azure-functions-host#4906]
#### Is your question related to a specific version? If so, please specify:
Azure Functions v2

#### What language does your question apply to? (e.g. C#, JavaScript, Java, All) 
C#

#### Question
Hey there, over on the Azure SDK team we are running into some problems using the new preview versions of our libraries within the Azure functions host because it overrides the version of ```System.Diagnostics.DiagnosticSource``` that we are using.

Specifically our libraries require access to the Activity.IdFormat property which is not present in the version of ```System.Diagnostics.DiagnosticSource``` which is loaded into the Azure Functions v2 host.

Wondering what plans you have - if any for upgrading.

Link to issue in the .NET repo for the Azure SDK where this dependency has been identified as an issue: https://github.com/Azure/azure-sdk-for-net/issues/7354

**Comment by brettsam** [comment]
We can give it a try once it's no longer preview. @fabiocav FYI.

**Comment by fabiocav** [comment]
@mitchdenny indeed, we typically adopt framework (.NET/ASP.NET) shortly after they're released, but depending on what you're looking for (is this for testing/validation only), we might be able to provide instructions on how to relax those restrictions.

**Comment by mitchdenny** [comment]
We ended up implementing a workaround on our side to get unblocked. Hopefully that'll hold until this package is released and you can update your dependencies.

**Comment by fabiocav** [comment]
@mitchdenny we've been working with @pakrym and we'll have that unblocked in the host soon. will keep this open to track.

**Comment by electricessence** [comment]
@mitchdenny what was your workaround?

**Comment by mitchdenny** [comment]
The work around was in the Azure.Core library itself. Basically some reflection to detect the capabilities of the underlying dependency and adjust accordingly.

**Comment by electricessence** [comment]
@mitchdenny I'm struggling with a problem with `System.Text.Json` having a sub dependency pinned. :(
Does anyone know a way to override the pinned reference?

**Comment by Alxandr** [comment]
Was this closed because it was resolved? I see the `Update diagnostic source to 4.7.0` issue is still open...

**Comment by mitchdenny** [comment]
Sorry I was in the GitHub mobile app. Must have accidentally tapped close instead of scrolling up.
